File: Beelina.APP/src/app/_models/product-warehouse-stock-receipt-entry.ts
import { IModelNode } from '../_interfaces/imodel-node';
import { Entity } from './entity.model';
import { ProductStockWarehouseAudit } from './product-stock-warehouse-audit';
import { ProductWarehouseStockReceiptDiscount } from './product-warehouse-stock-receipt-discount.model';
import { Supplier } from './supplier';
import { PurchaseOrderStatusEnum } from '../_enum/purchase-order-status.enum';
import { NumberFormatter } from '../_helpers/formatters/number-formatter.helper';

export class ProductWarehouseStockReceiptEntry extends Entity implements IModelNode {
  public supplierId: number;
  public supplier: Supplier;
  public stockEntryDate: Date;
  public referenceNo: string;
  public plateNo: string;
  public notes: string;
  public warehouseId: number;
  public discounts: Array<ProductWarehouseStockReceiptDiscount>;
  public invoiceNo: string;
  public invoiceDate: Date;
  public dateEncoded: Date;
  public purchaseOrderStatus: PurchaseOrderStatusEnum;
  public location: string;
  public productStockWarehouseAuditInputs: Array<ProductStockWarehouseAudit>;
  public grossAmount: number;
  public netAmount: number;

  get formattedGrossAmount(): string {
    return NumberFormatter.formatCurrency(this.grossAmount);
  }

  get formattedNetAmount(): string {
    return NumberFormatter.formatCurrency(this.netAmount);
  }

  constructor() {
    super();
    this.supplier = new Supplier();
    this.productStockWarehouseAuditInputs = [];
    this.discounts = [];
    this.invoiceNo = '';
    this.location = '';
    this.purchaseOrderStatus = PurchaseOrderStatusEnum.OPEN;
    this.grossAmount = 0;
    this.netAmount = 0;
  }
}

File: Beelina.LIB/Models/ProductWarehouseStockReceiptEntry.cs
using Beelina.LIB.Enums;
using Beelina.LIB.Interfaces;
using System.ComponentModel.DataAnnotations.Schema;

namespace Beelina.LIB.Models
{
    public class ProductWarehouseStockReceiptEntry
    : Entity, IUserActionTracker
    {
        public int? SupplierId { get; set; }
        public DateTime? StockEntryDate { get; set; }
        public string ReferenceNo { get; set; }
        public string PlateNo { get; set; }
        public string Notes { get; set; }
        public int WarehouseId { get; set; }
        public string InvoiceNo { get; set; }
        public DateTime? InvoiceDate { get; set; }
        public DateTime? DateEncoded { get; set; }
        public PurchaseOrderStatusEnum PurchaseOrderStatus { get; set; }
        public string Location { get; set; }

        public List<ProductStockWarehouseAudit> ProductStockWarehouseAudits { get; set; }
        public List<ProductWarehouseStockReceiptDiscount> Discounts { get; set; }
        public virtual Supplier Supplier { get; set; }

        [NotMapped]
        public double GrossAmount { get; set; }

        [NotMapped]
        public double NetAmount { get; set; }

        public int? DeletedById { get; set; }
        public virtual UserAccount DeletedBy { get; set; }
        public int? UpdatedById { get; set; }
        public virtual UserAccount UpdatedBy { get; set; }
        public int? CreatedById { get; set; }
        public virtual UserAccount CreatedBy { get; set; }
        public int? DeactivatedById { get; set; }
        public virtual UserAccount DeactivatedBy { get; set; }
    }
}